.github/workflows/main.yml

```diff
@@ -19,11 +20,245 @@ on:
+        docker run "${OPTS[@]}" \
+          cmake \
+            -Wdev -Werror=dev -Wdeprecated -Werror=deprecated \
+            -DDOXYGEN_EXECUTABLE=/usr/local/bin/doxygen \
+            -DWERROR=ON \
+            -DCOMPILE_DOCS=ON \
+            -DWITH_TESTS=OFF ..
+        docker run "${OPTS[@]}" make doc
+        docker run "${OPTS[@]}" make sphinx
+
+
+  # Linting 
+  lint:
+    runs-on: ubuntu-latest
+    needs: build-docker-ci
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v2
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Run clang-format
+      run: |
+        mkdir -p build-check-format
+
+        OPTS=(--rm --mount type=bind,src=$(pwd),target=/code/kotekan -w /code/kotekan/build-check-format/ ${IMG_CI}:${{ needs.build-docker-ci.outputs.branch_name }})
+        docker run "${OPTS[@]}" cmake -Wdev -Werror=dev -Wdeprecated -Werror=deprecated ..
+        docker run "${OPTS[@]}" make clang-format
+        git diff --exit-code
+
+    - name: Run black
+      run: |
+        OPTS=(--rm --mount type=bind,src=$(pwd),target=/code/kotekan -w /code/kotekan/ ${IMG_CI}:${{ needs.build-docker-ci.outputs.branch_name }})
+        docker run "${OPTS[@]}" black --exclude docs --check .
+
+    - name: Run cmakelint
+      run: |
+        OPTS=(--rm --mount type=bind,src=$(pwd),target=/code/kotekan -w /code/kotekan/ ${IMG_CI}:${{ needs.build-docker-ci.outputs.branch_name }})
+        docker run "${OPTS[@]}" tools/cmakelint.sh /code/kotekan
+
+
+  # Build a basic version of kotekan
+  build-base:
+    runs-on: ubuntu-latest
+    needs: build-docker-ci
+
+    steps:
+    - uses: actions/checkout@v3
+
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v2
+      with:
+        registry: ${{ env.REGISTRY }}
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Cache ccache files
+      uses: actions/cache@v3
+      with:
+        path: .ccache
+        key: ccache-base-build-${{ github.sha }}
+        restore-keys: |
+          ccache-base-build
+          ccache-full-build
+
+    - name: Build kotekan
+      run: |
+        OPTS=(--rm --mount type=bind,src=$(pwd),target=/code/kotekan -w /code/kotekan/build/ ${IMG_CI}:${{ needs.build-docker-ci.outputs.branch_name }})
+        docker run "${OPTS[@]}" ccache -s
+        docker run "${OPTS[@]}" \
+          cmake -Wdev -Werror=dev -Wdeprecated -Werror=deprecated \
+            -DWERROR=ON \
+            -DCMAKE_LINK_WHAT_YOU_USE=ON \
+            -DCMAKE_BUILD_TYPE=Test \
+            -DARCH=haswell \
+            -DCCACHE=ON \
+            -DWITH_TESTS=ON ..
+        docker run "${OPTS[@]}" make -j 2
+        docker run "${OPTS[@]}" ccache -s
+
+  # Build a full version of kotekan and run the unit tests
+  build-full-tests:
```

Comment:
I might suggest including the Ubuntu version number here as well.  While we can trace the dependencies, it's nice to get a message that tells you quickly "oh the 18.04 or 22.04 build is breaking".